multi_folder_download.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import os
+import subprocess
+import sys
+import time
+from datetime import datetime
+from distutils.util import strtobool
+
+import django
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos import GEOSGeometry
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_processing.settings')
+
+import geospaas_processing.downloaders as downloaders
+from geospaas.catalog.models import Dataset
+
+
+def main_function(ar):
+    if strtobool(ar.rel_time_flag):
+        designated_start_time = datetime.now().replace(tzinfo=tzutc())+ relativedelta(
+            hours=-int(ar.rel_time_lag))
+        designated_end_time = datetime.now().replace(tzinfo=tzutc())
+    else:
+        designated_start_time = datetime.strptime(ar.start_time, "%Y.%m.%d:%H").replace(
+            tzinfo=tzutc())
+        designated_end_time = datetime.strptime(ar.end_time, "%Y.%m.%d:%H").replace(tzinfo=tzutc())
+    downloaders.DownloadManager.MAX_DOWNLOADS = 500
+    download_manager = downloaders.DownloadManager(
+        download_directory=ar.mount_dir.rstrip(os.path.sep)+os.path.sep+ar.dir_struct.lstrip(
+            os.path.sep),
+        use_file_prefix=strtobool(ar.use_filename_prefix),
+        provider_settings_path=os.path.join(os.path.dirname(
+            __file__), ar.config_folder, 'provider_settings.yml'),
+        time_coverage_start__gte=designated_start_time,
+        time_coverage_end__lte=designated_end_time,
+        geographic_location__geometry__intersects=GEOSGeometry(ar.geometry),
+        dataseturi__uri__contains=ar.criteria)
+    download_manager.download()
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description='Process the arguments of entry_point (all must be in str)')
+    parser.add_argument('-md', '--mount_dir', required=True, type=str)
+    parser.add_argument('-cr', '--criteria', required=True, type=str)
+    parser.add_argument('-st', '--start_time', required=True, type=str)
+    parser.add_argument('-et', '--end_time', required=True, type=str)
+    parser.add_argument('-rtf', '--rel_time_flag', required=True, type=str)
+    parser.add_argument('-rtl', '--rel_time_lag', required=True, type=str)
```

Comment:
I think it is better to use just one begin date instaed of two:
either
`multi_folder_download.py -begin 2020-10-10 -end 2020-10-20 ... -other options`
or
`multi_folder_download.py -begin -10 --relative-time ... -other options`

So remove  `rel_time_lag` and convert `-rtf` to boolean (see action='store_true')

Reply:
As I have mentioned, since we use `dockerpy` we can not convey another type different from `str`. So, action='store_true' can't be implemented.

I think it is good to discriminate the input values from the very beginning in order to avoid confusion for the user. start_time should receive the date of start and rel_time_lag should receive hours! Since these two are two different things, it is better to receive them as two different arguments. What is your opinion @aperrin66 ?

Reply:
I agree with Anton. I find it more confusing to have two options for the same thing than having a flag modifying the behavior of an option. How can the user know what will happen if they specify both options in one call? With Anton's solution there is no ambiguity.

Also, I don't see why you couldn't use `action='store_true'`.

Reply:
@aperrin66 
@akorosov 
we use `dockerpy` somewhere in the deploy repo. So we need to pass all the arguments as a string! because we have to pass them as a list of strings to the dockerpy. This way, dockerpy can call the module with all its arguments. Otherwise it does not work and return the timeout error after 60 seconds. Because it cant parse the arguments that are sent to `client.containers.run()` command

Reply:
@aanersc , did you check how to use `action='store_true'` ?

It allows to have options like:

```python
my_script.py --foo

# in the script:
parser.add_argument('--foo', action='store_true')

args = parser.parse()
if args.foo:
    # do somtheing when script is called with --foo
else:
    # do somtheing when script is called without --foo
```

Reply:
Yeah, I know! :)
You will see in the next commit.